sql/src/test/java/io/crate/integrationtests/disruption/seqno/SimpleVersioningIT.java

```diff
@@ -0,0 +1,132 @@
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("delete from test where id = 1 and _seq_no = 10 and _primary_term = 1");
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("delete from test where id = 1 and _seq_no = 10 and _primary_term = 2");
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("delete from test where id = 1 and _seq_no = 1 and _primary_term = 2");
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("refresh table test");
+
+        for (int i = 0; i < 10; i++) {
+            execute("select _seq_no, _primary_term from test where id = 1");
+            assertThat(response.rows()[0][0], equalTo(1L));
+            assertThat(response.rows()[0][1], equalTo(1L));
+        }
+
+        // select with versioning
+        for (int i = 0; i < 10; i++) {
+            execute("select _seq_no, _primary_term, _version from test");
+            assertThat(response.rows()[0][0], equalTo(1L));
+            assertThat(response.rows()[0][1], equalTo(1L));
+            assertThat(response.rows()[0][2], equalTo(2L));
+        }
+
+        // select without versioning
+        for (int i = 0; i < 10; i++) {
+            execute("select _seq_no, _primary_term from test");
+            assertThat(response.rows()[0][0], equalTo(1L));
+            assertThat(response.rows()[0][1], equalTo(1L));
+        }
+
+        execute("delete from test where id = 1 and _seq_no = 1 and _primary_term = 1");
+        assertThat(response.rowCount(), equalTo(1L));
+
+    }
+
+    public void test_simple_version_with_refresh() throws Exception {
+        execute("create table test (id integer primary key, value string)");
+
+        ensureGreen();
+
+        execute("insert into test (id, value) values (?, ?) returning _seq_no", new Object[]{1, "value1_1"});
+
+        long seqNo = (long) response.rows()[0][0];
+        assertThat(seqNo, equalTo(0L));
+
+        execute("refresh table test");
+
+        execute("update test set value = 'value1_2' where id = 1 returning _seq_no, _primary_term");
+
+        seqNo = (long) response.rows()[0][0];
+        assertThat(seqNo, equalTo(1L));
+
+
+        execute("update test set value = 'value1_1' where id = 1 and _seq_no = 0 and _primary_term = 1 returning _seq_no, _primary_term");
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("delete from test where id = 1 and _seq_no = 0 and _primary_term = 1");
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("select _version from test where id=1");
+        assertThat(response.rows()[0][0], equalTo(2L));
+
+        execute("refresh table test");
+
+        execute("select _version, _seq_no from test");
+        assertThat(response.rows()[0][0], equalTo(2L));
+        assertThat(response.rows()[0][1], equalTo(1L));
```

Comment:
also this run 10 times: https://github.com/elastic/elasticsearch/blob/master/server/src/test/java/org/elasticsearch/versioning/SimpleVersioningIT.java#L316
Also not sure if that is really needed/adds additional scenarios to cover, any reasoning from your side to do or not do it?

Reply:
i think the main of running something 10 times is to proof that it is deterministic, i left out on this testcase because the operations seem really simple. What do you think ?

Reply:
i think the intention is to valide that reads from possible replicas behave correctly so i‘d keep the loop